qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1060,10 +1064,20 @@ def share_level(self, new_level: str) -> None:
                 specified. For example, IBM Quantum experiment service allows
                 "public", "hub", "group", "project", and "private".
         """
-        self._share_level = new_level
+        self._set_share_level_without_save(new_level)
```

Comment:
I think this can be removed and changed to overriding share_level setter in subclass as described further down

```suggestion
        self._share_level = new_level
```